packages/kit/src/exports/index.js

```diff
@@ -29,12 +29,23 @@ export function redirect(status, location) {
 export function json(data, init) {
 	// TODO deprecate this in favour of `Response.json` when it's
 	// more widely supported
+	const response = text(JSON.stringify(data), init);
+
+	if (!response.headers.has('content-type')) {
+		response.headers.set('content-type', 'application/json');
+	}
+
+	return response;
+}
+
+/** @type {import('@sveltejs/kit').text} */
+export function text(body, init) {
 	const headers = new Headers(init?.headers);
-	if (!headers.has('content-type')) {
-		headers.set('content-type', 'application/json');
+	if (!headers.has('content-length')) {
+		headers.set('content-length', String(body.length));
```

Comment:
This doesn't work right? Probably only works with ascii.

I had to do `Buffer.byteLength(str).toString()`

Reply:
good catch, thank you. the non-Node version of this is `new TextEncoder().encode(body).byteLength.toString()`